### PR TITLE
`na.rm` functionality for univariate utility statistics

### PR DESCRIPTION
**Pull request by jhseeman**
@awunderground few design questions before opening the full PR (still need to add tests based on responses to questions below)

* I added some functionality to handle `na.rm` and `drop_zeros` simultaneously - does this logic look good? 
* For proportions, I added functionality via `keep_empty_levels` that maintains unobserved levels throughout the proportion analysis - are there other places where we should turn this into a helper function? 
* I put the helper functions in `util_na_helper.R`, happy to move them if you have another preferred location

### CODE REVIEW

File: R/util_na_helper.R [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        paste(
+          "Some variables contain missing data: ",
+          paste(names(combined_data)[na_cols], collapse=", ")
+        )
+      )
+      
+      # stop if drop_zeros incompatible with keeping NAs
+      if (drop_zeros) {
+        
+        stop("Cannot set na.rm == FALSE and drop_zeros == TRUE with missing data")
```
Comment (awunderground) [inline R/util_na_helper.R:35]: Why is this? Because the `NA` could be `0` in reality?

Comment (jhseeman) [reply R/util_na_helper.R:35]: yeah - now that I think about it tho, we could specify an alternative aggregation function that handles the `drop_zeros` logic without the current strategy of converting 0s to NAs

**Comment by jhseeman** [comment]
> Maybe keep empty levels will be useful for disclosure risk evaluation?

100% the goal - this is the partition selection problem through a disclosure risk lens (ex: can I figure out reliably whether there is at least 1 person in the confidential dataset with x properties?)

File: R/util_proportions.R [hunk 2]
```diff
@@ -113,35 +118,131 @@ util_proportions <- function(postsynth,
...
-  combined_data <- combined_data %>%
+  combined_data_long <- combined_data %>%
     tidyr::pivot_longer(
       cols = -c(source, {{ group_by }}, ".temp_weight"), 
       names_to = "variable", 
       values_to = "class"
     ) 
   
+  # if flagged, join in empty levels to pivoted data
+  if (keep_empty_levels) {
```
Comment (awunderground) [inline R/util_proportions.R:149]: This section of code desperately needs some commenting. It is pretty tough to follow.

File: R/util_proportions.R [hunk 3]
```diff
@@ -113,35 +118,131 @@ util_proportions <- function(postsynth,
...
+    
+    prop_col_names <- combined_data %>% 
+      dplyr::select(-c(source, {{ group_by }}, ".temp_weight")) %>%
+      names
```
Comment (awunderground) [inline R/util_proportions.R:153]: `names()`

File: R/util_proportions.R [hunk 4]
```diff
@@ -113,35 +118,131 @@ util_proportions <- function(postsynth,
...
+          )
+        } else {
+          return(
+            data.frame(
+              "class" = levels(combined_data[[.x]])) %>% 
+              dplyr::mutate(
+                "variable" = .x
+              )  
+          )
+        }
```
Comment (awunderground) [inline R/util_proportions.R:174]: This may be too big for an anonymous function. Consider making a function inside of this function.

**Review by awunderground — APPROVED** [review]
The functionality looks good!

`util_proportions()` gets a little dense int he middle and would benefit from additional commenting.

**Comment by jhseeman** [comment]
Updated with the requested changes

**Pull request merged**